### ISSUE TITLE
multiple models on the same machine

### ISSUE THREAD

**Issue by boubakerwa** [pytransitions/transitions#251]
Hey everyone :) 
Great repo and great work :) 

I defined a machine and I have this issue. Whenever I call a "model"."transition"(), it gets called the number of models defined on the machine times. I checked the states of the other models and they don't change. 

Here's a part of my code: 
`light_one = Hue("hue one", device='1')
light_two = Hue("hue two", device='2')
#hue states and hue transitions have been already defined
hue_machine = Machine(states=hue_states, transitions=hue_transitions, initial = 'offstate')
hue_machine.add_model(light_one)
hue_machine.add_model(light_two)`

now the possible states of the hue_machine are: **Off, On** or **transiting** and one example that triggers this problem is calling the transition **turn_off** that goes from 'on' to 'off'.

so when the model is in 'on' state, calling light_one.turn_off() will call on_enter_off() twice.

I've spent a couple of hours trying to debug my code, but I think there's something I oversaw in the description or maybe there's a small issue with the machine :) 

Would love to hear from you :) 
Cheers,
Wassim

**Comment by aleneum** [comment]
Hello @boubakerwa,

its hard to tell without a working example but you might face this issue because your `Machine` instance is added as a model as well. If you do not explicitly pass a model during initialisation, this is the expected behaviour. In case you actually do not want that to happen either pass the models to the constructor or pass `model=None`.
```python
hue_machine = Machine(model=[light_one, light_two], states=hue_states, transitions=hue_transitions, initial = 'offstate')
```
If that does not solve your issue please provide a minimal working example for us to bugfix.

**Comment by boubakerwa** [comment]
Thanks for your answer :)
I tried to reproduce it using the example given in the tutorial and tried your solution but I think it's still not working. Here's the code:
`from transitions import Machine

class Matter():
    def on_enter_liquid(self):
	print("on_enter_liquid has been called")


states= ['solid', 'liquid', 'gas', 'plasma']
transitions = [
    { 'trigger': 'melt', 'source': 'solid', 'dest': 'liquid' },
    { 'trigger': 'evaporate', 'source': 'liquid', 'dest': 'gas' },
    { 'trigger': 'sublimate', 'source': 'solid', 'dest': 'gas' },
    { 'trigger': 'ionize', 'source': 'gas', 'dest': 'plasma' }
]

lump1 = Matter()
lump2 = Matter()

machine = Machine(model=[lump1, lump2], states=states, transitions=transitions, initial='solid', add_self=False)`
calling lump1.melt() prints the output twice.

I also tried this:
`from transitions import Machine

class Matter():
    def on_enter_liquid(self):
	print("HELLO")


states= ['solid', 'liquid', 'gas', 'plasma']
transitions = [
    { 'trigger': 'melt', 'source': 'solid', 'dest': 'liquid' },
    { 'trigger': 'evaporate', 'source': 'liquid', 'dest': 'gas' },
    { 'trigger': 'sublimate', 'source': 'solid', 'dest': 'gas' },
    { 'trigger': 'ionize', 'source': 'gas', 'dest': 'plasma' }
]

lump1 = Matter()
lump2 = Matter()

machine = Machine(model=None, states=states, transitions=transitions, initial='solid')

machine.add_model(lump1)
machine.add_model(lump2, initial='liquid')`
same problem.

Thanks for your time

**Comment by aleneum** [comment]
Sorry @boubakerwa,

but I cannot reproduce your observation:

```python
from transitions import Machine
from mock import MagicMock

class Matter(object):

    def __init__(self):
        self.mock = MagicMock()

    def on_enter_liquid(self):
        print("Called")
        self.mock()

states = ['solid', 'liquid', 'gas', 'plasma']
transitions = [
    {'trigger': 'melt', 'source': 'solid', 'dest': 'liquid'},
    {'trigger': 'evaporate', 'source': 'liquid', 'dest': 'gas'},
    {'trigger': 'sublimate', 'source': 'solid', 'dest': 'gas'},
    {'trigger': 'ionize', 'source': 'gas', 'dest': 'plasma'}
]

lump1 = Matter()
lump2 = Matter()

machine = Machine(model=[lump1, lump2], states=states, transitions=transitions, initial='solid')
lump1.melt()

assert lump1.mock.call_count == 1
assert lump2.mock.called is False

lump1 = Matter()
lump2 = Matter()

machine = Machine(model=None, states=states, transitions=transitions, initial='solid')
machine.add_model(lump1)
machine.add_model(lump2)

lump2.melt()

assert lump2.mock.call_count == 1
assert lump1.mock.called is False
```

This works just fine for me. Which version of transition are you using?

**Comment by boubakerwa** [comment]
Hum okay. I'm using transitions (0.5.3)

**Comment by boubakerwa** [comment]
Worked just fine after updating to 0.6.0! Thanks a lot! :) Next time I'll upgrade before asking a question :)

**Comment by aleneum** [comment]
I can confirm that this was a bug in `0.5.3` which was fixed as a side effect of #244.
